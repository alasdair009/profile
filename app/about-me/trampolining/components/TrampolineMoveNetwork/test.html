<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
