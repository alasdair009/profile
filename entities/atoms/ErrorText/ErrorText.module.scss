@use "../../../app/styles/colors";
@use "../../../app/styles/paragraph";

.root {
  @include paragraph.paragraphStyles;

  color: colors.$colorRedHeat;
  font-style: italic;
  display: block;
  text-align: var(--text-align);

  &Hide {
    display: none;
  }
}
