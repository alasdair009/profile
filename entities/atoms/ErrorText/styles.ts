"use client";

import { colors, Paragraph } from "@/entities";
import styled from "styled-components";

export const Root = styled(Paragraph)<{ $shown: boolean }>`
  color: ${colors.redHeat};
  font-style: italic;
  display: ${({ $shown }) => ($shown ? "block" : "none")};
`;
