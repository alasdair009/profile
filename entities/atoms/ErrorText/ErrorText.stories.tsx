import { ErrorText } from "./ErrorText";
import { Meta, StoryObj } from "@storybook/react";
import { expect, within } from "@storybook/test";

const meta: Meta<typeof ErrorText> = {
  component: ErrorText,
  args: {
    children: "Please fix the mistake.",
    shown: true,
  },
};
export default meta;

export const Default: StoryObj<typeof ErrorText> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const errorTextElement = canvas.getByTestId(ErrorText.name);

    await expect(errorTextElement).toBeInTheDocument();
    await expect(errorTextElement).toHaveTextContent(`${args.children}`);
  },
};
