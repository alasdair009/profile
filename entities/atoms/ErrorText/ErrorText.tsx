import { Root } from "./styles";
import { HTMLAttributes, ReactNode } from "react";
import {colors} from "@/entities";

type ErrorTextProps = {
    shown?: boolean;
} & HTMLAttributes<HTMLParagraphElement>;

export function ErrorText({ shown=true, ...rest }: ErrorTextProps) {
    return <Root color={colors.redHeat} as="span" fontSize="small" $shown={shown} {...rest} />;
}
