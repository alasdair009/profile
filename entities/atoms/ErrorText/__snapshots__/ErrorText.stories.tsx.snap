// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/ErrorText Default play-test 1`] = `
<span color="#df1c41"
      font-size="small"
      data-testid="ErrorText"
      class="sc-hDcvty dOVzab"
>
  Please fix the mistake.
</span>
`;
